### ISSUE TITLE
Internal mic switch commit

### ISSUE THREAD

**Issue by Radiomix2000** [jamesshao8/portapack-havoc-modified#12]
Hi, I noticed you added internal mic switch in commit https://github.com/jamesshao8/portapack-havoc-modified/commit/2759561b45c704911fadf76b4ad969a86653e428
I have Portapack H1. Where should I connect the internal microphone on PCB?

**Comment by jamesshao8** [comment]
Hi The ports I am using is the 5 port(power) and 6/7 port(signal input) of the AK4951. There are 6 welding points on the back of portapack. Three of them near the middle of the board are what I am talking about. You can find the 5 port by measuring voltage change when hitting the mic switch button on top of the ui. 6 and 7 is near 5. However  to really use the mic, it’s not enough by directly soldering a mic to it. You need a resistor same as r18 and a capacitor as c25. Solder r18 to port 5, and c25 to port 6/7, then solder the other side of them as a junction, this junction can now be used to connect with the mic (for powering it up and receiving the voice signal from the mic). The other side of the mic should be connected to the ground.

Please give me your testing result. As I am also testing it. Currently my result is the mic switch button can indeed switch the mic from 3.5mm head phone on and off. But when it’s off, would internal mic be switched on is not tested yet.

This function together with the speaker is important for the qrp app. With the speaker and microphone in place, it will be easier to use like a walker-talkie.

**Comment by antonlalala** [comment]
But can a fragment of the circuit ?! and then after google translator nothing is clear😁

**Comment by Radiomix2000** [comment]
![mic](https://user-images.githubusercontent.com/20892439/82142718-060f7e00-9847-11ea-8c33-1fdf68bc02d1.jpg)
This is the modification cirucuit diagram for internal mic use.

**Comment by jamesshao8** [comment]
@Radiomix2000 yes that’s what I am talking about. But I assume only tp4 and tp5 needed, no need to solder tp6. Need your confirmation.

**Comment by jamesshao8** [comment]
https://www.bilibili.com/video/bv16Q4y1A76r

Hi, I soldered the circuit and achieved what I was expecting. If everyone is interested for this feature, can give it a try.

**Comment by Radiomix2000** [comment]
Just checked it. Works, but has some bugs.
Two situations:
1.
The state at the beginning: Mic icon on upper bar is GREY.
When I enter to MIC app, and push the upper bar's MIC icon(it's changes to green now), then internal mic doesn't work. I need to push mic icon again(it changes back to grey), then internal mic works..

2.
The state at the beginning: Mic icon is grey.
If I just enter Mic app, do nothing, don't push mic icon, then internal mic doesn't work too.

**Comment by jamesshao8** [comment]
For the color, it's not really a bug. It is designed as grey for internal mic and green for 3.5mm mic.

Ideally, the internal mic should start when enter mictx. If it's not the case, it can be fixed later. My next step is designing a new pcb with internal mic and r18 c25. Once that down I will come back for the software modification.

**Comment by jamesshao8** [comment]
@Radiomix2000 where did you find this schematic? I didnt find on AK4951's datasheet.

**Comment by Radiomix2000** [comment]
@jamesshao8 Hi, it's here
https://github.com/furrtek/portapack-havoc/blob/master/hardware/portapack_h1/portapack_h1.pdf
Page nr. 2.
And thank you for your ideas for portapack!

**Comment by jamesshao8** [comment]
@Radiomix2000 Thanks. Are you interested in adding a battery indicator for portapack? I noticed in the code that rssi value is read from adc1 of hackrf, and hackrf has several spare adc port. If we can feed the voltage of portapack's battery to one of the adc port on hackrf, we will be able to show it's battery usage.

The software is not a problem. And I can modify the pcb of portapack, but I dont want to modify the pcb of hackrf. Is it possible to redefine a gpio port to transfer the battery information from portapack to hackrf adc?

**Comment by jamesshao8** [comment]
I am thinking of using the spare channel of mic input to measure the battery data. But may need some suggestion in analog cirsuits that connects the power and mic input.

**Comment by Radiomix2000** [comment]
@jamesshao8 Hi,
for me it's not so important, cause I used some additional PCBs to monitor battery charge/discharge. But for others the battery indicator could be  very useful.

**Comment by michaelbuud19** [comment]
> I am thinking of using the spare channel of mic input to measure the battery data. But may need some suggestion in analog cirsuits that connects the power and mic input.

Hello
Need last zip file for portapack to download
Latest firmware
How to contact you in telegram?

**Comment by jamesshao8** [comment]
I don't have a telegram. My email 13671890179@139.com

https://www.youtube.com/watch?v=6v2ie8el0po
